lib/fauna/client.rb

```diff
@@ -237,6 +237,8 @@ def perform_request(action, path, query, data)
         req.body = FaunaJson.to_json(data) unless data.nil?
         req.url(path || '')
       end
+    rescue Faraday::ConnectionFailed, Faraday::TimeoutError => e
```

Comment:
The other two errors under `Faraday::Error` that we'd get are `SSLError` and `ParsingError`. I don't think we can expect a status code to be returned in either case, so I'll just raise `UnavailableError` for everything but `ParsingError`